metric-observe-only/metric-observe-only.sh

```diff
@@ -93,13 +96,86 @@ do_measurement() {
     "ssh -o StrictHostKeyChecking=accept-new localhost true"
 
   # Retrieve measurement results
-  lxc file pull "$INSTNAME/home/ubuntu/results-first.json" "results-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-first.json"
-  lxc file pull "$INSTNAME/home/ubuntu/results-warm.json" "results-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-warm.json"
+  lxc file pull "$INSTNAME/home/ubuntu/results-first.json" "results-ssh-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-first.json"
+  lxc file pull "$INSTNAME/home/ubuntu/results-warm.json" "results-ssh-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-warm.json"
+}
+
+do_measurement_processcount() {
+  # Check how many processes are active after just booting
+  resultfile="results-processcount-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-$STAGE.txt"
+  Cexec ps -e --no-headers > "${resultfile}"
+}
+
+do_measurement_cpustat() {
+  # Check idle memory and cpu consumption after just booting
+  resultfile="results-cpustat-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-$STAGE.txt"
+  # We gather 3m avg + since boot
+  Cexec vmstat --one-header --wide --unit m 180 2 > "${resultfile}"
+}
+
+do_measurement_meminfo() {
+  # Check idle memory and cpu consumption after just booting
+  resultfile="results-meminfo-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-$STAGE.txt"
+  Cexec cat /proc/meminfo > "${resultfile}"
+}
+
+do_measurement_ports() {
+  resultfile="results-ports-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-$STAGE.txt"
+  Cexec ss -lntup > "${resultfile}"
+}
+
+do_measurement_disk() {
+  resultfile="results-disk-$RELEASE-$WHAT-c$CPU-m$MEM-$timestamp-$STAGE.txt"
+  Cexec df / --block-size=1M > "${resultfile}"
+}
+
+do_install_services() {
+  # This isn't very advanced, it installs various services in their default
+  # configuration to recheck if any of them changed their default behavior
+  # or footprint.
+  Cexec apt-get -qy install \
+      postgresql-all mysql-server \
+      libvirt-daemon-system containerd runc \
+      nfs-kernel-server samba \
+      slapd krb5-kdc sssd \
+      haproxy pacemaker \
+      memcached \
+      chrony \
+      nginx apache2 squid python3-django \
+      dovecot-imapd dovecot-pop3d postfix \
+      openvpn strongswan
 }
 
 cleanup
 setup_lxd_minimal_remote
 setup_container
 wait_load_settled
-do_measurement
+# Evict all caches after load settled
+Cexec sync
+Cexec dd of=/proc/sys/vm/drop_caches <<<'3'
+sleep 5s
+
+STAGE="early"
+do_measurement_cpustat
+do_measurement_meminfo
+do_measurement_ports
+do_measurement_processcount
+do_measurement_disk
+do_measurement_ssh
+
+do_install_services
```

Comment:
As you can imagine, I was already pondering/bikeshedding with myself on this before :-)
I couldn't come up with a much better name that wouldn't imply something else which we are not doing - especially if metric gathering is further extended in the future.

`baseline` is interesting, but feels just as wrong as many others I thought about.
But your question made me ponder about it again - I realized that (in the current form) there is something else than "observe only" that unites them. For doability in this cycle we did select the simple cases, so actually this should be `metric-simple.sh` now.

Furthermore I found in the jenkins that there `metric-foundations-...`, `metric-server-...`, `metric-openstack-...` and `metric-ssh-noninteractive-...` - I think we should correctly map this under server. Since this is renaming the ssh-noninteractive anyway I think a much better name would be:
`metric-server-simple`

Thanks for making me re-think again.

Reply:
Implemented, resolving conversation